nixopsgce/resources/gce_disk.py

```diff
@@ -96,14 +97,44 @@ def create(self, defn, check, allow_reboot, allow_recreate):
         if self.state != self.UP:
             extra_msg = ( " from snapshot '{0}'".format(defn.snapshot) if defn.snapshot
                      else " from image '{0}'".format(defn.image)       if defn.image
+                     else " marked as public. "                        if defn.public_image_project
                      else "" )
             self.log("creating GCE disk of {0} GiB{1}..."
                      .format(defn.size if defn.size else "auto", extra_msg))
+
+            # Retrieve GCENodeImage based on family name and project
+            if defn.public_image_project:
```

Comment:
hm I wonder if there is benefit from creating disks out of images ? afaik root disks are created automatically in the machine resource

Reply:
True but I did not change the current logic. I think it won't harm to leave it just in case someone wanted to create extra gceDisks from an image ( public or not ) given that it's technically possible to do that in GCP